rubix-core/src/main/java/com/qubole/rubix/core/CachedReadRequestChain.java

```diff
@@ -148,8 +154,8 @@ public Long call() throws IOException
       if (ex instanceof CancelledException) {
         throw ex;
       }
-
       log.error(String.format("Fall back to read from object store for %s .Could not read data from cached file : ", localCachedFile), ex);
+      customMetricReporterProvider.getMetricsCollectorInstance().addMetric(CachingFileSystemMetrics.DIRECT_READ_COUNT);
```

Comment:
should we not add metric when we fall back to direct read in case of NLRRC when exception occur?
@stagraqubole

Reply:
We fallback to remote from cachingInputStream too (check readFullyDirect)...

Reply:
ya, we should add the fallback count wherever we fallback

Reply:
Added 2 more metrics